tools/system_libs.py

```diff
@@ -1765,13 +1765,16 @@ def retrieve():
       except ImportError:
         try:
           from urllib.request import urlopen
-          f = urlopen(url)
-          data = f.read()
         except ImportError:
           # Python 2 compatibility
           from urllib2 import urlopen
-          f = urlopen(url)
-          data = f.read()
+        try:
+          import certifi
+          cafile = certifi.where()
+        except ImportError:
+          cafile = None
+        f = urlopen(url, cafile=cafile)
```

Comment:
IIUC this would be python2-incompatible since urlopen there doesn't have the `cafile` argument, but this bit is running unconditionally. I think we should either keep this section compatible or remove the existing compatibility bits.

Reply:
According to the docs it looks like it does take that argument: https://docs.python.org/2/library/urllib2.html ..

Oh.. but it looks like it was was added in 2.7.9.. is that what you mean?  Maybe I'll just leave this for now then until we actually remove the python2 support.

Reply:
oh you're right, i was just looking at the wrong page I think. So that's fine

Another question though, what does this PR actually solve?

Reply:
It solves for people that have `certifi` installed but not `requests`... which maybe is not a common use case.. so I will hold off.

Reply:
to be clear, I don't think this is necessarily bad, I had just never heard of `certifi`. Clearly someone must be in this situation though, what prompted you to do this PR in the first place?

Reply:
On macOS I was trying to do the smallest thing to make to certificates available.   In theory installing certifi I thing... but since I figured I would have use `pip` to install it anyway I might as well just install `requests` (which uses certifi internally) and call it done.

Reply:
ah, if certificates aren't wired up right, do we just get errors downloading the ports over HTTPS?

Reply:
Yeah, its been fairly persistent cause of bugs over the years.  Someone its only a problem on macOS where I guess the system doesn't include certificates in a form that python can consume out of the box.   

However the bundled version of python now includes the "requests" module so we should be good.